torch/nn/modules/rnn.py

```diff
@@ -507,6 +507,23 @@ class LSTM(RNNBase):
         All the weights and biases are initialized from :math:`\mathcal{U}(-\sqrt{k}, \sqrt{k})`
         where :math:`k = \frac{1}{\text{hidden\_size}}`
 
+    .. warning::
+        There are known deterministic issues for LSTM using cuDNN 7.6.5, 8.0 on CUDA 10.1 or later.
```

Comment:
`known non-determinism issues`. Is it for LSTM only, or RNN/GRU are also affected?

Reply:
Is the non-deterministic behavior really only on those two versions of cuDNN and only if the version of CUDA is 10.1 or later?

Reply:
It could be related to other RNN. I'll check that and add docs at other places if necessary.

Reply:
You may also want to cover yourself and say "On some versions of cuDNN and CUDA..." It's not great, since then people will never know if they may hit this issue or not, but it's better than telling them it may only happen in cases X and Y and then seeing it happen in case Z, too. 

Too bad there's no way to query for whether the function will be deterministic or not in the current environment, or request that it be run deterministically.

Reply:
I tested on cuda 10.2, cudnn 7.6.5. RNN and LSTM are affected. GRU is deterministic.